x/bridge/observer/osmosis/osmosis.go

```diff
@@ -0,0 +1,208 @@
+	msg := bridgetypes.NewMsgInboundTransfer(
+		in.Id,
+		in.Sender,
+		in.To,
+		bridgetypes.AssetID{
+			SourceChain: string(in.SrcChain),
+			Denom:       in.Asset,
+		},
+		math.Int(in.Amount),
+	)
+	fees := sdktypes.NewCoins(sdktypes.NewCoin(OsmoFeeDenom, OsmoFeeAmount))
+	bytes, err := o.osmoClient.SignTx(ctx, msg, fees, OsmoGasLimit)
+	if err != nil {
+		return errorsmod.Wrapf(err, "Failed to sign tx for inbound transfer %s", in.Id)
+	}
+	_, err = o.osmoClient.BroadcastTx(ctx, bytes)
+	if err != nil {
+		return errorsmod.Wrapf(
+			err,
+			"Failed to broadcast tx to Osmosis for inbound transfer %s",
+			in.Id,
+		)
+	}
+	return nil
+}
+
+func (o *ChainClient) observeEvents(
+	ctx context.Context,
+	txs <-chan coretypes.ResultEvent,
+) {
+	defer close(o.outboundChan)
+
+	eventToObserve := cosmosproto.MessageName(&bridgetypes.EventOutboundTransfer{})
+	for {
+		select {
+		case <-o.stopChan:
+			return
+		case event := <-txs:
+			newBlock, ok := event.Data.(comettypes.EventDataNewBlock)
+			if !ok {
+				continue
+			}
+
+			o.lastObservedHeight.Store(math.Max(
+				o.lastObservedHeight.Load(),
+				uint64(newBlock.Block.Height),
+			))
+			results, err := o.cometRpc.TxSearch(
+				ctx,
+				fmt.Sprintf("tx.height=%d", newBlock.Block.Height),
+				false,
+				nil,
+				nil,
+				"",
+			)
+			if err != nil {
+				o.logger.Error("Failed to fetch Txs at height ", newBlock.Block.Height)
+				continue
+			}
+
+			for _, r := range results.Txs {
+				if r.TxResult.IsErr() {
+					continue
+				}
+				for _, e := range r.TxResult.Events {
+					if e.Type != eventToObserve {
+						continue
+					}
+
+					out, err := outboundTransferFromEvent(uint64(r.Height), r.Hash.String(), e)
+					if err != nil {
+						continue
+					}
+					select {
+					case o.outboundChan <- out:
+					case <-o.stopChan:
+						o.logger.Info(
+							"Exiting early, event %s skipped in tx %s",
+							e.String(),
+							r.Hash.String(),
+						)
+						return
+					}
+				}
+			}
+		}
+	}
+}
+
+func outboundTransferFromEvent(height uint64, hash string, e abci.Event) (observer.Transfer, error) {
+	mes, err := sdktypes.ParseTypedEvent(e)
+	if err != nil {
+		return observer.Transfer{}, errorsmod.Wrapf(err, "Failed to parse typed event")
+	}
+	ev, ok := mes.(*bridgetypes.EventOutboundTransfer)
+	if !ok {
+		return observer.Transfer{}, fmt.Errorf("Failed to parse EventOutboundTransfer from event")
+	}
+
+	return observer.Transfer{
+		SrcChain: observer.ChainIdOsmosis,
+		DstChain: observer.ChainIdBitcoin,
+		Id:       hash,
+		Height:   height,
+		Sender:   ev.Sender,
+		To:       ev.DestAddr,
+		Asset:    ev.AssetId.Denom,
+		Amount:   math.Uint(ev.Amount),
+	}, nil
+}
+
+// Returns current height of the chain
+func (o *ChainClient) Height() (uint64, error) {
+	return o.lastObservedHeight.Load(), nil
+}
+
+// Returns number of required tx confirmations
+func (o *ChainClient) ConfirmationsRequired() (uint64, error) {
+	// Query bridge module
+	return 0, nil
+}
```

Comment:
This is going to change right?

Reply:
I think it is still returning zero in at least one of the methods as per https://github.com/osmosis-labs/osmosis/pull/7882#discussion_r1544438144

Reply:
No, I got it a little mixed up. I left a comment for the Bitcoin client thinking it was Osmosis (ConfirmationsRequired does not need to be called in Bitcoin), but here everything is ok now